### ISSUE TITLE
:nth-of-type(2n - 1) space in argument results in error

### ISSUE THREAD

**Issue by kkopachev** [sass/libsass#1650]
Updated to new libsass/node-sass recently and noticed regression (or intended?) with the code:
```
tr:nth-of-type(2n - 1) { color: red; }
```
outputs an error
> "Invalid CSS after "": expected 1 selector or at-rule, was "th-of-type(4n -...""

While code with no spaces works:
```
tr:nth-of-type(2n-1) { color: red; } // works!
tr:nth-of-type(2n + 1) { color: red; } // works too!
tr:nth-of-type(2n+1) { color: red; } // and this one is fine
```

spacing around minus sign makes libsass complain.
> $ node-sass -v
node-sass	3.4.1	(Wrapper)	[JavaScript]
libsass  	3.3.1	(Sass Compiler)	[C/C++]

**Comment by mgreter** [comment]
OK, this is pretty odd.

```
:nth-of-type(2n-1),
:nth-of-type(2n-  1),
:nth-of-type(2n  -1),
:nth-of-type(2n  -  1),
:nth-of-type( 2n  -  1 )
{ color: red; }
:nth-of-type(2n+1),
:nth-of-type(2n+  1),
:nth-of-type(2n  +1),
:nth-of-type(2n  +  1),
:nth-of-type( 2n  +  1 )
{ color: red; }
```

In compressed mode this produces:
```
:nth-of-type(2n-1),
:nth-of-type(2n-  1),
:nth-of-type(2n  -1),
:nth-of-type(2n  -  1),
:nth-of-type(2n  -  1)
{color:red}

:nth-of-type(2n+1),
:nth-of-type(2n+1),
:nth-of-type(2n+1),
:nth-of-type(2n+1),
:nth-of-type(2n+1)
{color:red}  
```

No idea why ruby sass does not apply the same logic here. Looks like a bug to me ...

**Comment by xzyfer** [comment]
The white space being preserved looks like the work of a static value parser.

**Comment by xzyfer** [comment]
Looks like Ruby Sass' `almost_any_value_token` method is the cause.

**Comment by chriseppstein** [comment]
That's a ruby bug, not a high priority one, but definitely a bug.

**Comment by mgreter** [comment]
@chriseppstein thanks for the confirmation, I guess it's a one byte fix on the ruby side ...

**Comment by mgreter** [comment]
This should be fixed on master for release 3.3.3.
